gm4_liquid_tanks/data/liquid_tanks/functions/liquid_value_update.mcfunction

```diff
@@ -0,0 +1,23 @@
+#@s = liquid tank
+#run from liquid_tanks:process
+
+#set values
+scoreboard players operation @s gm4_lt_value += @s gm4_lt_buffer
+scoreboard players set @s gm4_lt_buffer 0
+scoreboard players operation @s gm4_lt_value < @s gm4_lt_max
+
+#if value of 0, set to empty
+execute if score @s gm4_lt_value matches ..0 at @s run function liquid_tanks:empty
+
+#calculating teleport distance
+scoreboard players operation last_liquid_height gm4_lt_disp_val = @s gm4_lt_disp_val
+scoreboard players operation liquid_height_calc gm4_lt_disp_val = @s gm4_lt_value
+scoreboard players set 10 gm4_lt_disp_val 10
+scoreboard players operation liquid_height_calc gm4_lt_disp_val *= 10 gm4_lt_disp_val
+scoreboard players operation liquid_height_calc gm4_lt_disp_val /= @s gm4_lt_max
+scoreboard players operation @s gm4_lt_disp_val = liquid_height_calc gm4_lt_disp_val
+scoreboard players operation liquid_height_calc gm4_lt_disp_val -= last_liquid_height gm4_lt_disp_val
+
+#teleport proper number of times.
+execute if score liquid_height_calc gm4_lt_disp_val matches 1.. at @s run function liquid_tanks:teleport_display_up
+execute if score liquid_height_calc gm4_lt_disp_val matches ..-1 at @s run function liquid_tanks:teleport_display_down
```

Comment:
I suggest moving the score into the other armorstand. This way `teleport_display_up` and `teleport_display_down` don't have to constantly do an `execute as`

Reply:
I meant to move the score at the end of the calculation. It’s just temporary.

Reply:
Oh I see. In that case, I don't think it would gain any benefit. Copying the score would require at least one selector resolution, and then changing the executor would be another. Currently there is just one in the actual teleport command. It is very rare that the armor stand need to teleport more than one step up or down, so on most every occasion, the selector would only be resolved once.